### PR TITLE
Update Binaryen for 2020

### PR DESCRIPTION
**Pull request by dcodeIO**
In celebration of the new year, this updates our Binaryen bindings to include fireworks.

- [x] Update bindings
- [x] Attempt to avoid versioning issues on the web
- [x] Integrate new reference types features (ref.null, ref.is_null, ref.func)

### CODE REVIEW

**Comment by dcodeIO** [comment]
Last commit also adds a little SDK to use on the web, primarily targeting WebAssembly Studio, that glues asc to the specific Binaryen version it requires. If that works as expected, we can get rid of conflicts by using the SDK in studio's setup code.

**Comment by dcodeIO** [comment]
What's left is `ref.func`, which we don't have a good story for yet. For instance, we can't do

```js
var anyref = someFunc;
```

yet due to `someFunc` being represented by a function table index, while `ref.func` wants a static function index (respectively function name in the Binaryen API). Perhaps we can compile that differently, will take a look.

**Comment by dcodeIO** [comment]
Ok, that turned out to be easier than expected :)

File: lib/sdk/tests/index.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js"></script>
+<script>
+require(
+  ["../index"],
+  function(sdk) {
+    console.log(sdk);
```
Comment (MaxGraey) [inline lib/sdk/tests/index.html]: I guess better print only versions.
```ts
const { BINARYEN_VERSION, ASSEMBLYSCRIPT_VERSION } = sdk;
console.log(
`Bynaryen v${BINARYEN_VERSION}
AssemblyScript v${ASSEMBLYSCRIPT_VERSION}`
);
```
wdyt?

Comment (dcodeIO) [reply lib/sdk/tests/index.html]: This is mostly there so one can inspect the result in a browser, including the contents of the objects. It's easy to see if something went wrong then, like if something is an empty object or doesn't have the contents one would expect.

Comment (MaxGraey) [reply lib/sdk/tests/index.html]: got it

Comment (dcodeIO) [reply lib/sdk/tests/index.html]: Made it print both, plus a few assertions :)

**Pull request merged**